### PR TITLE
Add `signup` page form

### LINKED ISSUE #910: Sign up form
This issue is part of the epic #855

Add a sign up form to allow users to create a new account
The path should be `/signup`

### LINKED ISSUE #911: Add Facebook authentication
This issue is part of the epic #855 

Add Atlas App Services Facebook Authentication: https://www.mongodb.com/docs/atlas/app-services/authentication/facebook/

To integrate into a React App:
https://www.mongodb.com/docs/realm/web/authenticate/#use-the-built-in-facebook-oauth-2.0-flow

### LINKED ISSUE #927: Add Google Authentication
This issue is part of the epic https://github.com/responsible-ai-collaborative/aiid/issues/855

Add Atlas App Services Google Authentication: https://www.mongodb.com/docs/atlas/app-services/authentication/google/

To integrate into a React App:
https://www.mongodb.com/docs/realm/web/authenticate/#use-the-built-in-google-oauth-2.0-flow

### PR DESCRIPTION
**Pull request by pdcp1**
This PR adds a user sign up form on `/signup`

- [x] Login with email and password. Closes #910
- [x] Login with Facebook. Closes #911
- [x] Login with Google. Closes #927 

image

### 👉 [Preview testing link](https://feature-sign-up-form--pablocosta-staging.netlify.app/)

It's related to the epic issue #855

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -473,6 +473,35 @@ REALM_GRAPHQL_API_KEY=xxxxxxxxxx
 ```
 About Realm API Keys: https://www.mongodb.com/docs/realm/authentication/api-key/
 
+## Facebook login integration
```
Comment (cesarvarela) [inline README.md:476]: Looks like the title should mention Google too, or the more generic term "social login"

File: README.md [hunk 2]
```diff
@@ -473,6 +473,35 @@ REALM_GRAPHQL_API_KEY=xxxxxxxxxx
...
+
+To allow users to login with Facebook, you will need to add the following configuration to your Atlas App Service.
```
Comment (cesarvarela) [inline README.md:478]: Maybe here too, something like "to enable social logins..."

File: README.md [hunk 3]
```diff
@@ -473,6 +473,35 @@ REALM_GRAPHQL_API_KEY=xxxxxxxxxx
...
+facebookAppId = [Facebook App ID, see comment below for more information]
+googleClientId = [Google Client ID, see comment below for more information]
+loginRedirectUri = [Login Authentication redirect URI, see comment below for more information]
+```
+
+- To get the Facebook App ID you should go to the [Facebook Developer Portal](https://developers.facebook.com/apps/), and check your app.
+- In order to get the Google Client ID (OAuth 2.0 Client ID) you should set up an OAuth 2.0 following these [instructions](https://support.google.com/cloud/answer/6158849?hl=en). After set it up, you can find the Google Client ID in your [Google Cloud Console](https://console.cloud.google.com/apis/credentials).
+- Login Authentication redirect URI is the URL that the user will be redirected to after successfully authenticating with Facebook or Google. It should point to `/logincallback` page. For Production the URI is `https://incidentdatabase.ai/logincallback`
+
+Also add the following secret value to your Atlas App Service following the instructions in the [Atlas App Services documentation](https://www.mongodb.com/docs/atlas/app-services/values-and-secrets/define-and-manage-secrets/).
```
Comment (cesarvarela) [inline README.md:492]: should say `values` I think

File: site/gatsby-site/i18n/locales/es/login.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  "Sign up with Facebook": "Registrarse con Facebook",
+  "Sign up with Google": "Registrarse con Google",
+  "Logged in as ": "Ingresado como ",
+  "Logging you out...": "Saliendo...",
+  "Succesfully sent password reset email": "Se ha enviado un correo para restablecer la contraseña",
+  "Account created": "Cuenta creada",
+  "Enter new password": "Ingrese la nueva contraseña",
+  "Confirm new password": "Confirme la nueva contraseña",
+  "Succesfully updated password, click here to ": "Se ha actualizado la contraseña, haga clic aquí para ",
+  "Don't have an account?": "¿No tiene una cuenta?",
```
Comment (cesarvarela) [inline site/gatsby-site/i18n/locales/es/login.json:19]: Not sure if we should use formal or informal language (yo uso tuteo 🤔 )

File: site/gatsby-site/i18n/locales/es/translation.json [hunk 5]
```diff
@@ -75,6 +74,19 @@
...
   "Read More": "Leer Mas",
+  "Email address": "Correo electrónico",
+  "or": "o",
+  "An unknown error has ocurred": "Ha ocurrido un error desconocido",
+  "Invalid email": "Correo electrónico inválido",
+  "Passwords must match": "Las contraseñas deben coincidir",
+  "Required": "Requerido",
+  "Password must be at least 6 characters long": "La contraseña debe tener al menos 6 caracteres",
+  "Password must be less than 128 characters long": "La contraseña debe tener menos de 128 caracteres",
+  "user not found": "usuario no encontrado",
```
Comment (cesarvarela) [inline site/gatsby-site/i18n/locales/es/translation.json:85]: Lowercase looks weird 🤔 

Any reason to not use capital case?

File: site/gatsby-site/src/contexts/userContext/UserContextProvider.js [hunk 6]
```diff
@@ -44,27 +46,60 @@ export const UserContextProvider = ({ children }) => {
...
-      credentials = Realm.Credentials.emailPassword(email, password);
-    } else {
-      credentials = Realm.Credentials.anonymous();
+      const user = await realmApp.logIn(credentials);
+
+      if (user.id === realmApp.currentUser.id) {
+        setUser(user);
+      }
+    } catch (e) {
+      console.error(e);
```
Comment (cesarvarela) [inline site/gatsby-site/src/contexts/userContext/UserContextProvider.js:85]: I think we can delete this and later implement: https://github.com/responsible-ai-collaborative/aiid/issues/586

**Review by cesarvarela — CHANGES_REQUESTED** [review]
Left some comments, where can I test this?

File: site/gatsby-site/i18n/locales/es/translation.json (hunk 5 above)
Comment (pdcp1) [reply site/gatsby-site/i18n/locales/es/translation.json:85]: Yes, that looks weird. The reason is that Realm returns the error messages without capital case and I want to translate them to display in the error toast message.

File: site/gatsby-site/i18n/locales/es/login.json (hunk 4 above)
Comment (pdcp1) [reply site/gatsby-site/i18n/locales/es/login.json:19]: I think tuteo is fine if we all agree on that.

File: site/gatsby-site/i18n/locales/es/translation.json (hunk 5 above)
Comment (pdcp1) [reply site/gatsby-site/i18n/locales/es/translation.json:85]: I added `tw-capitalize` tailwind class on each toast error message to avoid this.

**Comment by pdcp1** [comment]
> Left some comments, where can I test this?

@cesarvarela To test this feature please use this Preview link https://feature-sign-up-form--pablocosta-staging.netlify.app/

**Comment by cesarvarela** [comment]
Got this when using Facebook login:

image

**Comment by pdcp1** [comment]
@cesarvarela The Facebook App is in development mode until we finally launch this feature. By now, I added you as an App Admin user on the FB App. Please accept the invitation and test again.

**Comment by smcgregor** [comment]
Do both the Facebook and Google auth methods give us an email address to anchor identity against?

**Comment by cesarvarela** [comment]
I think showing a spinner before the logged in state is known, can prevent the login form from flashing, if for example you are already logged in.

**Comment by pdcp1** [comment]
> Do both the Facebook and Google auth methods give us an email address to anchor identity against?

@smcgregor Yes, both give us an email address. It's stored in the Atlas App Service (Realm) in the same way as the default Email/Pass sign up.

image

**Comment by pdcp1** [comment]
> I think showing a spinner before the logged in state is known, can prevent the login form from flashing, if for example you are already logged in.

@cesarvarela That's a good idea. I will add a spinner

**Comment by pdcp1** [comment]
@cesarvarela This is ready to review. Now it has a loading spinner on Mail, Facebook and Google login and sign up forms

**Comment by cesarvarela** [comment]
I sill this this:

https://user-images.githubusercontent.com/1172479/185681223-64befb23-962f-4584-a3c8-48b684fcb00a.mov


Happens when you refresh the page.

**Comment by pdcp1** [comment]
@cesarvarela loading state fixed, please review

**Comment by cesarvarela** [comment]
This might have been broken from before but the forgot password form seems it is not working:

image

**Comment by cesarvarela** [comment]
I've used the facebook login:

image

**Review by cesarvarela — COMMENTED** [review]
We'll be merging this after the current production deploy is out

**Comment by pdcp1** [comment]
> Maybe because I used the facebook login?

@cesarvarela that's correct, the forgot/reset password functionality only works for Email/Password authentication, not for Facebook or Google authentication.

**Review by cesarvarela — APPROVED** [review]

**Pull request merged**